### DISCUSSION TITLE
Silicon Labs EFR32MG24 (Silabs EFR32 MG24) SoCs and EFR32xG24 dev kits officially announced

### DISCUSSION THREAD

**Discussion by Hedda** [zigpy/zigpy#893] [General]
This news comes close after release of [Silicon Labs Gecko SDK (GSDK) 4.0 and Silabs Zigbee EmberZNet SDK 7.0](https://github.com/zigpy/zigpy/discussions/879) published a few weeks ago.

Silicon Labs has now announced the new EFR32MG24 (EFR32 MG24) SoCs and Silabs EFR32xG24 development kits + radio boards:

https://news.silabs.com/2022-01-24-Silicon-Labs-Brings-AI-and-Machine-Learning-to-the-Edge-with-Matter-Ready-Platform

https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-socs

https://www.cnx-software.com/2022/01/24/silicon-labs-bg24-and-mg24-2-4-ghz-wireless-mcus-quadruple-ai-performance-at-a-fraction-of-the-energy/

Looking at specs MG24 won't make a much better Zigbee coordinator for end-users compåared to the MG21 (but maybe a better router / end devices?), at least not unless have several hundreds of Zigbee 3.0 devices (like 200+ Zigbee 3.0 devices in total) or will use a single firmware that has both native Zigbee stack and native Thread/Matter stack running simultaneously/concurrently locally at the same time?

* EFR32MG24 come with much higher RAM memory, with 128 or 256 KB variants, (compared to 64 or 96 KB variants of EFR32MG21).
* EFR32MG24 flash storage has been increased to 1536 KB or 1024 KB, (compared 512, 768, or 1024 KB in the EFR32MG21).
* Interestingly their similar ARM Cortex-M33 core runs at 78 MHz compared the 80 MHz core in the EFR32MG21
* New feature is an[ integrated AI/ML accelerator NPU for edge interference](http://silabs.com/ai-ml) which can potentially lower devices power consumption. 
* _Silabs EFR32BG24 and EFR32MG24 SoCs are already shipping today to Alpha customers, with mass deployment slated for April 2022._

New "EFR32xG24 Pro Kit +10 dBm" + "EFR32xG24 Pro Kit +20 dBm" development boards and EFR32xG24 radio boards coming coon:

![image](https://user-images.githubusercontent.com/6320001/150969225-5fbd81fe-0af2-4c90-99ba-af0f4d81c426.png)

![image](https://user-images.githubusercontent.com/6320001/150969296-0b48fc8c-f401-401f-b7c6-ff3a87e58e5d.png)

No news on MGM240 (MGM240P and MGM240L) modules based on EFR32MG24 precertified with FCC/CE to replace MGM210.

https://www.silabs.com/wireless/zigbee/efr32mg21-series-2-modules

https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-modules

PS: At the same time they also announced EFR32BG24 (Silabs EFR32 BG24) which basically is the same chip locked to Bluetooth only.

https://www.silabs.com/wireless/bluetooth/efr32bg24-series-2-socs

**Comment by edpgcooper** [comment]
Hoping the MGM240P comes along soon and has Japanese/Giteki approvals, need a module with decent amount of flash with these approvals.

**Reply by Hedda** [reply]
FYI, Silabs has posted a data sheet and user manual for MGM240P model:

https://www.silabs.com/documents/public/data-sheets/mgm240p-datasheet.pdf

Also see FCC reports:

https://fccid.io/QOQ-GM240P

https://fcc.report/FCC-ID/QOQ-GM240P/

Other models of MGM240P and MGM240L with unknown specifications will be:

MGM240PA22VNA
MGM240PA32VNA
MGM240PA32VNN
MGM240PB22VNA
MGM240PB32VNA
MGM240PB32VNN

MGM240L022RNF
MGM240L022VNF

**Reply by edpgcooper** [reply]
Hasn’t that been out for a little while, it’s only got 32K of RAM though, so no good for my Thread/BLE plans.

**Reply by MattWestb** [reply]
8 dBm  output power and good receiver so OK for testing but not as one RCP for Zigbee and OTBR.

**Reply by wazpoppin** [reply]
@Hedda @edpgcooper MGM240P was just released. You can order the evaluation boards today from Digikey: XGM240-RB4316A (10dBm) & XGM240-RB4317A (20dBm)

**Reply by Hedda** [reply]
> MGM240P was just released. You can order the evaluation boards today from Digikey: XGM240-RB4316A (10dBm) & XGM240-RB4317A (20dBm)

Yeah, MGM240 (MGM240P and MGM240L) radio modules that are meant to be used in commercial products now look to be available:

https://www.silabs.com/wireless/zigbee/efr32mg24-series-2-modules

https://fccid.io/QOQ-GM240P

https://fcc.report/FCC-ID/QOQ-GM240P/

https://www.silabs.com/documents/public/data-sheets/mgm240p-datasheet.pdf

<html>
<body>


Part Number | MCU | Core Frequency (MHz) | Flash | RAM | Cryptography | Secure Vault | Output Power Range (dBm) | GPIO | Temperature Range ºC | Package Type | Antenna | Package Size (mm)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
MGM240PA22VNA  | ARM Cortex-M33 | 78 | 1536 | 256 | AES-128 AES-256  ECC  SHA-1  SHA-2 | Mid | -24  10 | 26 | -40  105 | PCB | Built-in | 12.9x15.0x2.2
MGM240PA32VNA | ARM Cortex-M33 | 78 | 1536 | 256 | AES-128 AES-256  ECC  SHA-1  SHA-2 | Mid | -33.7  19.9 | 26 | -40  105 | PCB | Built-in | 12.9x15.0x2.2
MGM240PA32VNN | ARM Cortex-M33 | 78 | 1536 | 256 | AES-128 AES-256  ECC  SHA-1  SHA-2 | Mid | -33.7  19.9 | 26 | -40  105 | PCB | RF-Pin | 12.9x15.0x2.2
MGM240PB22VNA| ARM Cortex-M33 | 78 | 1536 | 256 | AES-128 AES-256  ECC  SHA-1  SHA-2 | High | -24  10 | 26 | -40  105 | PCB | Built-in | 12.9x15.0x2.2
MGM240PB32VNA | ARM Cortex-M33 | 78 | 1536 | 256 | AES-128 AES-256  ECC  SHA-1  SHA-2 | High | -33.7  19.9 | 26 | -40  105 | PCB | Built-in | 12.9x15.0x2.2
MGM240PB32VNN | ARM Cortex-M33 | 78 | 1536 | 256 | AES-128 AES-256  ECC  SHA-1  SHA-2 | High | -33.7  19.9 | 26 | -40  105 | PCB | RF-Pin | 12.9x15.0x2.2


</body>
</html>

As you write, they have now also released those evaluation radio boards based on them that are meant to be used with Silicon Labs development boards from their starter kits: 

https://www.trustedparts.com/en/part/silicon-labs/XGM240-RB4316A

xGM240P radio module board = XGM240-RB4317A (20dBm).

https://www.silabs.com/development-tools/wireless/xgm240-rb4317a-xgm240p-module-radio-board

xGM240L radio module board = XGM240-RB4316A (10dBm).

https://www.silabs.com/development-tools/wireless/xgm240-rb4316a-xgm240p-module-radio-board

PS: Noted that those new radio module boards were expected to be called MGM240P and MGM240L but look like they will instead be marketed as Silabs xGM240x Series 2 Modules, specifically "xGM240P Broad-based Module" and "xGM240L Lighting Module".

**Comment by wazpoppin** [comment]
There is also BGM variant of the 240P radio module: https://www.silabs.com/documents/public/data-sheets/bgm240p-datasheet.pdf 

MGM can do both BLE and 802.15.4 while the BGM variant is BLE only. xGM240P/L most likely means that the development tools are the "superset" part i.e. the MGM240x, but they can be also be used to develop BLE only applications..

**Comment by qm3ster** [comment]
> unless have several hundreds of Zigbee 3.0 devices

That's a big unless.
People are definitely starting to feel the limitation by now.

**Reply by MattWestb** [reply]
Zigbee (3) networks with updated link key can have max 0xffff (64K) devices if the coordinator can storing the keys for every device but if using hashed link key as ZHA and Z2M is doing with EZSP its 0xffff device without hardware problem if the network is balanced and not having spamming devices that is taking it down or configure not standard broadcast settings that is blocking the router functionality in the network (Z2M TI firmware for the moment).

**Comment by Hedda** [comment]
FYI, SparkFun Electronics releasad a MGM240P based "**_SparkFun Thing Plus Matter_**" (DEV-20270) development board for $24.95(US):

https://www.sparkfun.com/products/20270

https://eu.mouser.com/new/sparkfun/sparkfun-mgm240p-dev-board/

https://www.mouser.se/ProductDetail/SparkFun/DEV-20270?qs=Jm2GQyTW%2FbhYWMD4vAe2Fg%3D%3D

https://www.iot-now.com/2023/03/06/128385-silicon-labs-sparkfun-electronics-reveal-thing-plus-matter-board-to-support-interoperability/

https://www.youtube.com/watch?v=TO45V9c-yFw&ab_channel=SparkFunElectronics

https://www.youtube.com/watch?v=POVRRuVCr-Q&ab_channel=SparkFunElectronics

https://www.youtube.com/watch?v=QtY499N2AFY&ab_channel=SparkFunElectronics

**Reply by MattWestb** [reply]
XG28 chips is star shipped and shall being more Matter friendly but i have not looking if some dev-board is being released with  them.
Not mass series is XG22, 23 and i think they was ending some more before making production runs for the chips.
